webapp/pages/settings/my-email-address/index.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <ds-card centered v-if="success">
+    <transition name="ds-transition-fade">
+      <sweetalert-icon icon="info" />
+    </transition>
+    <ds-text v-html="submitMessage" />
+  </ds-card>
+  <ds-form v-else v-model="form" :schema="formSchema" @submit="submit">
+    <template slot-scope="{ errors }">
+      <ds-card :header="$t('settings.email.name')">
+        <ds-input id="email" model="email" icon="at" :label="$t('settings.email.labelEmail')" />
+
+        <template slot="footer">
+          <ds-space class="backendErrors" v-if="backendErrors">
+            <ds-text align="center" bold color="danger">{{ backendErrors.message }}</ds-text>
+          </ds-space>
+          <ds-button class="submit-button" icon="check" :disabled="errors" type="submit" primary>
+            {{ $t('actions.save') }}
+          </ds-button>
+        </template>
+      </ds-card>
+    </template>
+  </ds-form>
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import { AddEmailAddressMutation } from '~/graphql/EmailAddress.js'
+import { SweetalertIcon } from 'vue-sweetalert-icons'
+
+export default {
+  components: {
+    SweetalertIcon,
+  },
+  data() {
+    return {
+      backendErrors: null,
+      success: false,
+    }
+  },
+  computed: {
+    submitMessage() {
+      const { email } = this.formData
+      return this.$t('settings.email.submitted', { email })
+    },
+    ...mapGetters({
+      currentUser: 'auth/user',
+    }),
+    form: {
+      get: function() {
+        const { email } = this.currentUser
+        return { email }
+      },
+      set: function(formData) {
+        this.formData = formData
+      },
+    },
+    formSchema() {
+      const { email } = this.currentUser
+      const sameEmailValidationError = this.$t('settings.email.validation.same-email')
+      return {
+        email: [
+          { type: 'email', required: true },
+          {
+            validator(rule, value, callback, source, options) {
+              const errors = []
+              if (email === value) {
+                errors.push(sameEmailValidationError)
+              }
+              return errors
+            },
+          },
+        ],
+      }
+    },
+  },
+  methods: {
+    async submit() {
+      const { email } = this.formData
+      try {
+        await this.$apollo.mutate({
+          mutation: AddEmailAddressMutation,
+          variables: { email },
+        })
+        this.$toast.success(this.$t('settings.email.success'))
+        this.success = true
+
+        setTimeout(() => {
+          this.$router.push({
+            path: 'my-email-address/enter-nonce',
+            query: { email },
+          })
+        }, 3000)
```

Comment:
I don't like that we are mixing different types of alerts – toasts, the error in the footer, the `success` block in combination with this timeout... they're all over the place! And in this case not really self-contained when the `submit()` method has to decide how long the alert is displayed. 😕 

Do you think we could agree on just one version, make a proper component for it and then use it everywhere? :smile: Maybe not in this PR, though...

Reply:
Yes, totally agree. I would prefer form validation errors. I try my best to implement re-usable async form validators, e.g. the one that checks for unique slugs.

However in this case we have no backend query to check if a particular already exists. Sure - I could probably implement a backend query for frontend validation. But I thought maybe we don't want to do that? :thinking: 

So right now, we have to call the mutation and create the user account. If it fails, we have to check the backend error message. It would be great to get the form validation error into the `<ds-form>` component somehow, but I believe that's not possible right now.

Reply:
I also prefer form validators as it's closer to where the user's attention is, but then, as far as I know, we'd have `toastrs` for success and form validators for failures?

Reply:
We have toasters because we're just to lazy to write the template and the translations for error messages.